cups/snmp.c

```diff
@@ -1285,6 +1287,8 @@ asn1_get_length(unsigned char **buffer,	/* IO - Pointer in buffer */
       length = (length << 8) | **buffer;
   }
 
+  if (length > bufend - *buffer)
```

Comment:
Since all of the other functions check that `*buffer` is less than `bufend`, I'm not sure this is necessary, either...

Reply:
Please see above for explanation. It could be that one of these checks can be omitted because the other is in place. Since I am not confident enough to see every possible combination, I preferred to have too many checks in place.